src/caf/base/data_structures.py

```diff
@@ -2191,26 +2230,59 @@ def check_compatibility(targets: Collection[IpfTarget], adjust: bool = False):
             Whether to change the targets or just report on their compatibility.
         """
         targ_dict = {i: j for i, j in enumerate(targets)}
+        # add reference to end of targ_dict if exists
+        if reference is not None:
+            targ_dict[len(targets)] = reference
```

Comment:
You're using a dict but treating it like a list by just using integer index values for the keys, I don't think it makes a real difference here but is just slightly confusing